Source/IC.cpp

```diff
@@ -11,6 +11,35 @@ Constant::Constant()
     for (int n = 0; n < velocity.size(); n++) {
         m_op.velocity[n] = velocity[n];
     }
+
+    std::string m_model_type;
+    pp.query("model_type", m_model_type); // ns: default is "isothermal".
+                                          // "energyD3Q27" activates product
+                                          // equilibrium, energy equation etc.
+
+    if (m_model_type == "energyD3Q27") {
+        m_op.m_model_type = 1;
+        amrex::Vector<amrex::Real> mach_components{AMREX_D_DECL(0, 0, 0)};
+        pp.queryarr("MachComponents", mach_components, 0, AMREX_SPACEDIM);
+        for (int n = 0; n < mach_components.size(); n++) {
+            m_op.MachComponents[n] = mach_components[n];
+        }
+
+        pp.query(
+            "initialTemperature",
```

Comment:
for user queries, use snake case everywhere. E.g here it would be `initial_temperature`